xbmc/guilib/GUIPanelContainer.cpp

```diff
@@ -399,9 +399,17 @@ void CGUIPanelContainer::ValidateOffset()
 
 void CGUIPanelContainer::SetCursor(int cursor)
 {
+  // exceeds the number of items the panel can hold
   if (cursor > m_itemsPerPage * m_itemsPerRow - 1)
     cursor = m_itemsPerPage * m_itemsPerRow - 1;
-  if (cursor < 0) cursor = 0;
+
+  // exceeds the number of items being displayed
+  if (cursor > static_cast<int>(m_items.size()) - 1 - GetOffset() * m_itemsPerRow)
```

Comment:
You are doing the same calculation two times in a row. Don’t you think you should calculate once and store the result in a variable?

Reply:
Right, initially while testing I was using a temporary variable, but in the end I changed it. I will change it